go/websocketbridge/main.go

```diff
@@ -0,0 +1,133 @@
+package websocketbridge
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Originate/exocom/go/exorelay"
+	"github.com/Originate/exocom/go/structs"
+	uuid "github.com/satori/go.uuid"
+
+	"golang.org/x/net/websocket"
+)
+
+// Config contains the configuration values for ExoRelay
+type Config struct {
+	Host string
+	Port int
+	Role string
+}
+
+// WebsocketBridge handles communication between an Exosphere backend and
+// a browser front-end using websockets
+type WebsocketBridge struct {
+	ClientPort int
+	Config     Config
+	server     http.Server
+	SocketMap  map[string]*websocket.Conn
+	exoRelay   exorelay.ExoRelay
+}
+
+// ConnectToExocom sets up an ExoRelay to communicate with exocom
+func (w *WebsocketBridge) ConnectToExocom() error {
+	w.exoRelay = exorelay.ExoRelay{
+		Config: exorelay.Config{
+			Host: w.Config.Host,
+			Port: w.Config.Port,
+			Role: w.Config.Role,
+		},
+	}
+	w.SocketMap = make(map[string]*websocket.Conn)
+	err := w.exoRelay.Connect()
+	if err != nil {
+		return err
+	}
+	messageChannel := w.exoRelay.GetMessageChannel()
+	go func() {
+		for {
+			message, ok := <-messageChannel
+			if !ok {
+				break // channel closed
+			}
+			socket := w.SocketMap[message.SessionID]
+			message.SessionID = ""
+			serializedBytes, err := json.Marshal(message)
+			if err != nil {
+				break
+			}
+			err = websocket.Message.Send(socket, serializedBytes)
+			if err != nil {
+				break
+			}
+		}
+	}()
+	return nil
+}
+
+// ListenForClients sets up a server to listen for incoming websocket connections
+func (w *WebsocketBridge) ListenForClients() error {
+	w.server = http.Server{
+		Handler: websocket.Handler(w.websocketHandler),
+		Addr:    fmt.Sprintf(":%d", w.ClientPort),
+	}
+	go w.server.ListenAndServe()
+	return nil
+}
+
+// Reset closes all the open sockets and nils the SocketMap
+func (w *WebsocketBridge) Reset() error {
```

Comment:
I don't think we will ever actually Reset a websocket bridge in production. I'd suggest creating a close instead. For the tests we can close created websocket bridges and create a new one for each test.